### PR TITLE
[Documents] Refactor page preview generation to queue in Page::save()

### LINKED ISSUE #11091: [Improvement]: Add page preview generation to queue in Page::save()
### Improvement description

Follow-up to: #11023
It's not necessary to call another endpoint for adding the page to the queue. Moving this to Page::save() also has the benefit that previews are updated even when updating the page via SDK.

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #11091

### CODE REVIEW

File: models/Document/Page.php [hunk 1]
```diff
@@ -274,4 +274,18 @@ public function getPreviewImageFilesystemPath()
...
+        // Dispatch page preview message, if preview is enabled.
+        $documentsConfig = \Pimcore\Config::getSystemConfiguration('documents');
+        if ($documentsConfig['generate_preview'] ?? false) {
+
+            \Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
+                new GeneratePagePreviewMessage($this->getId(), \Pimcore\Tool::getHostUrl())
+            );
+        }
+
+        return parent::save();
```
Comment (blankse) [inline models/Document/Page.php:289]: Shouldn't it be sent to the messenger bus after saving? Not that the old page is still generated?

Comment (blankse) [inline models/Document/Page.php]: Shouldn't it be sent to the messenger bus after saving? Not that the old page is still generated?
```suggestion
        parent::save();

        // Dispatch page preview message, if preview is enabled.
        $documentsConfig = \Pimcore\Config::getSystemConfiguration('documents');
        if ($documentsConfig['generate_preview'] ?? false) {
            \Pimcore::getContainer()->get('messenger.bus.pimcore-core')->dispatch(
                new GeneratePagePreviewMessage($this->getId(), \Pimcore\Tool::getHostUrl())
            );
        }

        return $this;
```

Comment (jdreesen) [reply models/Document/Page.php]: What about doing this in an `DocumentEvents::POST_ADD`/`DocumentEvents::POST_UPDATE` event listener instead?

Comment (dvesh3) [reply models/Document/Page.php]: @blankse is right. We should generate preview after calling parent::save() 👍. 

@jdreesen I'd not create a new listener for this task.

**Pull request merged**